### DISCUSSION TITLE
GTS 4 Mini App Dev

### DISCUSSION THREAD

**Discussion by Cotrox** [zepp-health/zeppos-forum#68] [Q&A]
Hi guys! I'm trying to use VSCode Zepp Extension to create my first App for Amazfit GTS 4 Mini. When i select the device to make a preview QR Code I can only choose GT3 Devices... why? How can I select GTS 4 Mini?

**Comment by juan518munoz** [comment]
You need to add the device to the `app.json` here's an example:
https://github.com/juan518munoz/ZeppOS-Spotify/blob/master/app.json

**Reply by Cotrox** [reply]
Ok I did it but when i click 'Preview' and scan the QR Code i receive error: null

It is just on an Hello World besic Test

**Reply by Cotrox** [reply]
The error is: /node_modules/@zeppos/device-types/index.d.ts probably... but i just generate an Hello World Project so... why?

**Reply by juan518munoz** [reply]
When doing `zeus preview`, does it make you choose the `target` device you're building for? Also check that you are logged in with `zeus login`

**Reply by Cotrox** [reply]
Well... i'm logged in.

Now i'm trying to restart from zero.

- Created the Hello World Project.
- (I've not installed Node.js ... can be the problem.. right?)
- Anyway when i go to app.json i can see the config of gtr3 devices and teide device. I delete them and put gts4-mini config.
- But i still see gtr pages in pages folder and not shows gt4 mini. Is that right?

**Reply by juan518munoz** [reply]
If the command `zeus` returns the list of commands available then you don't need to install anything else.

Editing the json file won't change any other file, the gtr pages will remain and gts 4 mini won't be generated.

If you want to test that the `zeus cli` works you can clone the repo I mentioned before, it has support for the gts 4 mini so you should be able to install it via qr code.

**Reply by Cotrox** [reply]
Ok it works! No way man! Spotify App! AHAHAH it is very nice! But it says no connection when i put on spotify on my iPhone. Anyway now I'll try to start from it!

**Reply by juan518munoz** [reply]
The app doesn't work because it doesn't have the credentials needed baked on the code you compiled, if you want to test a working version I can provide you with a preview.

Glad to help you, good luck on your coding journey.

**Reply by Cotrox** [reply]
If you need a tester i'll be glad ^_^

Anyway i'm still having problems when I generate a new app. The problem is in the jsconfig.json file:

```
"files": [
    "node_modules/@zeppos/device-types/index.d.ts"
  ],
```

It seams not working for these lines

**Reply by Cotrox** [reply]
Ok generating an empty app I've not this problem!

**Reply by Cotrox** [reply]
@juan518munoz let me know if you need tester for your ZeppOS Apps ^_^

**Reply by juan518munoz** [reply]
If you have a spotify premium account email me with your account's email and the device you plan to use.
juanmunoz890@gmail.com

**Reply by Cotrox** [reply]
> If you have a spotify premium account email me with your account's email and the device you plan to use.
> 
> juanmunoz890@gmail.com
> 
> 
> 
> 

Unfortunately i've not a premium account but I use GTS 4 Mini. Is not possibile?
